tests/test_util.py

```diff
@@ -137,3 +137,35 @@ def test_get_device_type_not_existing(tmpdir):
     with pytest.raises(EnvironmentError) as excinfo:
         _util.get_device_type(str(filename))
     pytest.assert_env_error(excinfo.value, errno.ENOENT, str(filename))
+
+
+@pytest.mark.skipif(sys.version_info >= (3,5),
+                    reason="system calls automatically restarted in python3.5")
+def test_eintr_retry_call(tmpdir):
+    import os, signal, select
+
+    def handle_alarm(signum, frame):
+        pass
+    orig_alarm = signal.getsignal(signal.SIGALRM)
+
+    # Open an empty file and use it to wait for exceptions which should
+    # never happen
+    filename = tmpdir.join('test')
+    filename.ensure(file=True)
+    fd = os.open(str(filename), os.O_RDONLY)
+
+    try:
+        signal.signal(signal.SIGALRM, handle_alarm)
+
+        # Ensure that a signal raises EINTR
+        with pytest.raises(select.error) as e:
+            signal.alarm(1)
+            select.select([], [], [fd], 2)
+
+        # Ensure that wrapping the call does not raise EINTR
+        signal.alarm(1)
+        assert _util.eintr_retry_call(select.select, [], [], [3], 2) == ([], [], [])
+    finally:
+        os.close(fd)
+        signal.signal(signal.SIGALRM, orig_alarm)
+
```

Comment:
Can you fiddle your testing so that it covers all Python versions? Perhaps add a new test for version >= 3.5 or else modify this test somehow.

Also, the convention for skipif is to make a str of the condition, like: str('sys.version_info >= (3,5)'). The seemingly redundant conversion is due to the future import of unicode strings.